libs/src/services/discoveryProvider/index.js

```diff
@@ -46,17 +47,21 @@ class DiscoveryProvider {
           whitelistMap[urlJoin(url, '/version')] = url
         })
 
-        let resp = await Utils.raceRequests(Object.keys(whitelistMap), (url) => {
-          pick = whitelistMap[url]
-        }, {})
-
-        const isValid = pick && resp.data.service && (resp.data.service === serviceType.DISCOVERY_PROVIDER)
-        if (isValid) {
-          console.info('Initial discovery provider was valid')
-          endpoint = pick
-        } else {
-          console.info('Initial discovery provider was invalid, searching for a new one')
-          endpoint = await this.ethContracts.selectDiscoveryProvider(this.whitelist)
+        try {
+          let resp = await Utils.raceRequests(Object.keys(whitelistMap), (url) => {
+            pick = whitelistMap[url]
+          }, {})
+
+          isValid = pick && resp.data.service && (resp.data.service === serviceType.DISCOVERY_PROVIDER)
+          if (isValid) {
+            console.info('Initial discovery provider was valid')
+            endpoint = pick
+          } else {
+            console.info('Initial discovery provider was invalid, searching for a new one')
+            endpoint = await this.ethContracts.selectDiscoveryProvider(this.whitelist)
+          }
+        } catch (e) {
+          console.error('Could not select a discprov from the whitelist', e)
```

Comment:
so this will just error and maybe behave weirdly, perhaps it should re-throw?

Reply:
good call. i just updated this